reconcile/prometheus_rules_validator.py

```diff
@@ -0,0 +1,54 @@
+import sys
+import semver
+import logging
+
+import utils.gql as gql
+import utils.threaded as threaded
+import utils.promtool as promtool
+import reconcile.openshift_resources_base as orb
+import reconcile.queries as queries
+
+QONTRACT_INTEGRATION = 'prometheus_rules_validator'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def check_rule(rule):
+    try:
+        promtool.check_rule(rule['spec'])
+    except Exception as e:
+        return {'path': rule['path'], 'message': str(e).replace('\n', '')}
+
+
+def run(dry_run, thread_pool_size=10, cluster_name=None):
+    orb.QONTRACT_INTEGRATION = QONTRACT_INTEGRATION
+    orb.QONTRACT_INTEGRATION_VERSION = QONTRACT_INTEGRATION_VERSION
+
+    gqlapi = gql.get_api()
+    rules_paths = queries.get_prometheus_rules_paths()
+
+    rules = []
+    for n in gqlapi.query(orb.NAMESPACES_QUERY)['namespaces']:
+        if n['name'] != 'openshift-customer-monitoring':
```

Comment:
why this artificial check? we can filter by

```
managedResourceTypes:    
- PrometheusRule    
- ...
```

Reply:
Because we don't have rules anywhere else... but I like your condition more. I will need to adjust a bit the error messages too.

Reply:
fixed in 210abfb